lib/cext/preprocess.rb

```diff
@@ -93,10 +93,32 @@
   rb_tr_release_handle(this->decoder_for_get_copy_data);
 EOF
 
+NOKOGIRI_DEALLOC_DECL_ORIG = <<-EOF
+static int dealloc_node_i(xmlNodePtr key, xmlNodePtr node, xmlDocPtr doc)
+{
+EOF
+
+NOKOGIRI_DEALLOC_DECL_NEW = <<-EOF
+static int dealloc_node_i(st_data_t a, st_data_t b, st_data_t c, int errorState)
+{
+  xmlNodePtr key = (xmlNodePtr)a;
+  xmlNodePtr node = (xmlNodePtr)b;
+  xmlDocPtr doc = (xmlDocPtr)c;
```

Comment:
Is this to avoid warnings turning into errors?
Maybe we can adjust the flags, like in openssl's extconf.rb, but patching might still be best to catch potential problems early.

Reply:
Yes, they play a little fast and loose with the function signature and when I was producing the patch set I wanted to be careful we didn't trip up on things like that.